python/lsst/ip/diffim/dipoleFitTask.py

```diff
@@ -859,43 +860,58 @@ def fitDipole(self, source, tol=1e-7, rel_weight=0.1,
             fp = source.getFootprint()
             self.displayFitResults(fp, fitResult)
 
-        fitParams = fitResult.best_values
-        if fitParams['flux'] <= 1.:   # usually around 0.1 -- the minimum flux allowed -- i.e. bad fit.
+        # usually around 0.1 -- the minimum flux allowed -- i.e. bad fit.
+        if fitResult.params['flux'].value <= 1.:
             return None, fitResult
 
-        centroid = ((fitParams['xcenPos'] + fitParams['xcenNeg']) / 2.,
-                    (fitParams['ycenPos'] + fitParams['ycenNeg']) / 2.)
-        dx, dy = fitParams['xcenPos'] - fitParams['xcenNeg'], fitParams['ycenPos'] - fitParams['ycenNeg']
+        # TODO: We could include covariances, which could be derived from
+        # `fitResult.params[name].correl`, but those are correlations.
+        posCentroid = measBase.CentroidResult(fitResult.params['xcenPos'].value,
+                                              fitResult.params['ycenPos'].value,
+                                              fitResult.params['xcenPos'].stderr,
+                                              fitResult.params['ycenPos'].stderr)
+        negCentroid = measBase.CentroidResult(fitResult.params['xcenNeg'].value,
+                                              fitResult.params['ycenNeg'].value,
+                                              fitResult.params['xcenNeg'].stderr,
+                                              fitResult.params['ycenNeg'].stderr)
+        centroid = measBase.CentroidResult((fitResult.params['xcenPos'] + fitResult.params['xcenNeg']) / 2,
+                                           (fitResult.params['ycenPos'] + fitResult.params['ycenNeg']) / 2.,
+                                           math.sqrt(posCentroid.xErr**2 + negCentroid.xErr**2),
```

Comment:
incorrect error propagation (see e.g. [here](https://en.wikipedia.org/wiki/Propagation_of_uncertainty#Example_formulae))

```suggestion
                                           math.sqrt(posCentroid.xErr**2 + negCentroid.xErr**2)/2,
```

Reply:
If you can access `covar` (have to confirm indices):

```suggestion
                                           math.sqrt(posCentroid.xErr**2 + negCentroid.xErr**2 + 2*fitResult.covar[0,1])/2.,
```

Reply:
I'm not totally surprised the uncertainties end up smaller: we're fitting 3 images instead of just one as in SdssCentroid, so there's more information available to the fitter.  But we can investigate further on DM-50839.

Reply:
In particular, the total centroid uncertainty (0.0088) is now less than the individual positive/negative uncertainties (0.013)! That may be correct, but it's still surprising to me.